### PR TITLE
Kubelet#GetNode Gets nodes instead of listing them

### LINKED ISSUE #13476: Kubelet lists all nodes instead of Get'ing node
`Kubelet#GetNode` invokes nodeLister.List(), and also the node cache should be using Nodes().Get instead of Nodes.List()  Not sure why the code has a comment that says it can't get a single item (because it can).

### PR DESCRIPTION
**Pull request by robertabbott**
This contribution is under the corporate CLA signed by Square Inc.

resolves #13476

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -713,17 +717,11 @@ func (kl *Kubelet) GetNode() (*api.Node, error) {
 	if kl.standaloneMode {
 		return nil, errors.New("no node entry for kubelet in standalone mode")
 	}
-	l, err := kl.nodeLister.List()
+	l, err := kl.nodeLister.GetNodeInfo(kl.nodeName)
```
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: You can just `return kl.nodeLister.GetNodeInfo(kl.nodeName)`

**Comment by yujuhong** [comment]
@k8s-bot ok to test

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (robertabbott) [inline pkg/kubelet/kubelet.go]: done

**Comment by yujuhong** [comment]
Thanks for the PR. LGTM!

**Pull request merged**